src/components/CollaboratorModal/CollaboratorModalContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { CollaboratorModalState } from "components/CollaboratorModal/constants"
+import { createContext, useContext, Dispatch, SetStateAction } from "react"
+
+interface CollaboratorModalProps {
+  newCollaboratorEmail: string
+  setNewCollaboratorEmail: Dispatch<SetStateAction<string>>
+  addCollaboratorError: string
+  setAddCollaboratorError: Dispatch<SetStateAction<string>>
+  modalState: CollaboratorModalState
+  setModalState: Dispatch<SetStateAction<CollaboratorModalState>>
+  isAcknowledged: boolean
+  setIsAcknowledged: Dispatch<SetStateAction<boolean>>
+  collaboratorData: any // TODO
+  collaboratorRoleData: { role: string } // TODO
+  addCollaborator: any // TODO
+  deleteCollaborator: any // TODO
+  handleAddCollaborator: any // TODO
+  handleDeleteCollaborator: any // TODO
+  isModalOpen: boolean
+  closeModal: () => void
+  localUser: any // TODO
+  deleteCollaboratorTarget: any // TODO
+  setDeleteCollaboratorTarget: Dispatch<SetStateAction<any>> // TODO
+}
```

Comment:
Just wondering, what are the benefits of introducing a context for this component? The amount of state management here seems to be quite large, but the Collaborator component only has 2 layers of components (all submodals are children of the parent CollaboratorModal), and quite a few pieces managed by the context are only used in a single submodal. I'm wondering if it's better to pass these as props directly to the subcomponents.

Would be happy to chat more about this offline - it's quite a big change, and I think we might want to do it in a separate PR if we want to go ahead with it

Reply:
My initial thought was that by using React context, we could make the `CollaboratorModal.tsx` cleaner because we can simply describe the selected child component without needing to specify the props to pass down as such:

```
// To render the selected modal
{renderModalContent(modalState)}
```

```
// The render selector function
  const renderModalContent = (currModalState: CollaboratorModalState) => {
    switch (currModalState) {
      case CollaboratorModalState.Acknowledgement:
        return <AcknowledgementSubmodal />
      case CollaboratorModalState.Default:
        return <MainSubmodal />
      case CollaboratorModalState.RemoveCollaborator:
        return <RemoveCollaboratorSubmodal />

      default:
        return <MainSubmodal />
    }
  }
```

Reply:
I don't have terribly strong feelings though - if we prefer not using Context in this case, I'm more than happy to undo it

Reply:
As discussed today during Isomer sync, we'll fix this as a separate PR after our 31 Oct deadline, since it is a non-functional blocking refactor.